s3/s3_operations.cpp

```diff
@@ -431,7 +431,7 @@ namespace irods_s3 {
                 (oflag & O_ACCMODE) == O_WRONLY, (oflag & O_ACCMODE) == O_RDWR, (oflag & O_ACCMODE) == O_RDONLY,
                 (oflag & O_TRUNC) != 0, (oflag & O_CREAT) != 0, (oflag & O_APPEND) != 0);
 
-        ios_base::openmode mode = 0;
+        ios_base::openmode mode = static_cast<ios_base::openmode>(0);
```

Comment:
This is fine, but I'm just now noticing we don't guard against invalid values for `oflag` (i.e. invalid access mode).

This can also be handled in a separate PR.

Reply:
Should I open an issue for it?